packet.cpp

```diff
@@ -0,0 +1,133 @@
+#include "packet.h"
+
+Packet::Packet(char data[24], tcp::socket& socket,
+               const std::shared_ptr<KeyValueStore>& k)
+    : bod_len(readUInt32LE(data, 8)),
+      opaque(readUInt32LE(data, 12)),
+      key_len(readUInt16LE(data, 2)),
+      op(data[1]),
+      ext_len(data[4]) {
+  read(socket);
+  if (data[1] == static_cast<char>(OpCode::GET)) {
+    if (k->has(key)) {
+      respondToGet(socket, k->get(key), true);
+    } else {
+      std::string msg = "Key not found";
+      std::vector<char> val(msg.cbegin(), msg.cend());
```

Comment:
Vector is being used a lot, maybe a ```using std::vector``` at the top? It's common enough people will know where it came from.